geode-lucene/src/main/java/org/apache/geode/cache/lucene/internal/repository/serializer/HeterogeneousLuceneSerializer.java

```diff
@@ -93,8 +70,16 @@ private void addSerializersForPrimitiveValues() {
   /**
    * Get the field mapper based on the type of the given object.
    */
-  private LuceneSerializer getFieldMapper(Object value) {
+  private LuceneSerializer getFieldMapper(Object value, String[] indexedFields) {
+    if (Arrays.asList(indexedFields).contains(LuceneService.REGION_VALUE_FIELD)) {
```

Comment:
This seems really expensive to do for every update. I think we need to come up with a more efficient way of delegating to the right mapper.